### PR TITLE
New Read-only cache FPGA Feature Tutorial

### PR DESCRIPTION
**Pull request by mohammadfawaz**
## Description

Creating a new tutorial that describes the usage of the read-only cache feature.

## Checklist
Administrative
- [x] Review sample design with the appropriate [Domain Expert](https://github.com/oneapi-src/oneAPI-samples/wiki/Reviewers-and-Domain-Experts): Mike Tucker

Code Development
- [x] Implement coding guidelines and ensure code quality. [see wiki for details](https://github.com/oneapi-src/oneAPI-samples/wiki/General-Code-Guidelines)
- [x] Adhere to readme template 
- [x] Enforce format via clang-format config file
- [x] Adhere to sample.json specification. https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-specification
- [x] Ensure/create CI test configurations for sample (ciTests field) https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-ci-test-object
- [x] Run jsonlint on sample.json to verify json syntax. www.jsonlint.com

Security and Legal
- [ ] OSPDT Approval (see @JoeOster for assistance)

Review
- [ ] Review DPC++ code with Paul Peterseon. (GitHub User: pmpeter1)
- [ ] Review readme with Tom Lenth(@tomlenth) and/or Joe Oster(@JoeOster)

### CODE REVIEW

**Review by mdbtucker — DISMISSED** [review]
I reviewed this code before this PR was created, looks like a good tutorial thanks Mohammad.

File: DirectProgramming/DPC++FPGA/Tutorials/Features/read_only_cache/src/read_only_cache.cpp [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+#include <sycl/ext/intel/fpga_extensions.hpp>
+
+#include <chrono>
+
+// dpc_common.hpp can be found in the dev-utilities include folder.
+// e.g., $ONEAPI_ROOT/dev-utilities//include/dpc_common.hpp
+#include "dpc_common.hpp"
+
+using namespace sycl;
+namespace oa = sycl::ext::oneapi;
```
Comment (pmpeter1) [inline DirectProgramming/DPC++FPGA/Tutorials/Features/read_only_cache/src/read_only_cache.cpp:16]: The oa seems a little too terse for a sample.     maybe ext_oneapi?   That shortens it by 7 characters and still highlights more explicitly it's a oneAPI extension feature.

File: DirectProgramming/DPC++FPGA/Tutorials/Features/read_only_cache/src/read_only_cache.cpp [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+                   "system has a correctly configured FPGA board.\n";
+      std::cerr << "Run sys_check in the oneAPI root directory to verify.\n";
+      std::cerr << "If you are targeting the FPGA emulator, compile with "
+                   "-DFPGA_EMULATOR.\n";
+    }
+    std::terminate();
+  }
+
+  // Compute the reference solution
+  uint16_t index = 0xFFFu;
```
Comment (pmpeter1) [inline DirectProgramming/DPC++FPGA/Tutorials/Features/read_only_cache/src/read_only_cache.cpp:124]: Why the magic number 0xFFFu?     A comment or better explanation would help.

**Review by pmpeter1 — CHANGES_REQUESTED** [review]

File: DirectProgramming/DPC++FPGA/Tutorials/Features/read_only_cache/src/read_only_cache.cpp (hunk 1 above)
Comment (mohammadfawaz) [reply DirectProgramming/DPC++FPGA/Tutorials/Features/read_only_cache/src/read_only_cache.cpp]: Addressed

File: DirectProgramming/DPC++FPGA/Tutorials/Features/read_only_cache/src/read_only_cache.cpp (hunk 2 above)
Comment (mohammadfawaz) [reply DirectProgramming/DPC++FPGA/Tutorials/Features/read_only_cache/src/read_only_cache.cpp]: Addressed

**Review by pmpeter1 — DISMISSED** [review]

**Review by tomlenth — APPROVED** [review]
looks good, thanks

**Review by mdbtucker — APPROVED** [review]
Re-approving after target branch was updated.

**Review by mdbtucker — APPROVED** [review]

**Pull request merged**